lib/classes/Variables.test.js

```diff
@@ -2378,6 +2378,42 @@ module.exports = {
     });
   });
 
+  describe('#getValueStrToBool()', () => {
+    beforeEach(() => {
+      serverless.variables.service = {
+        service: 'testService',
+        provider: serverless.service.provider,
+      };
+      serverless.variables.loadVariableSyntax();
+    });
+    it('should return true (boolean) for a true (string)', () => {
+      return serverless.variables.getValueStrToBool('strToBool(true)').should.become(true);
+    });
+    it('should return false (boolean) for a false (string)', () => {
+      return serverless.variables.getValueStrToBool('strToBool(false)').should.become(false);
+    });
+    it('should return true (boolean) for a truthy string', () => {
+      return serverless.variables.getValueStrToBool('strToBool(anything)').should.become(true);
+    });
+    it('should throw error for a null-like string', () => {
+      return serverless.variables.getValueStrToBool('strToBool(null)').should.become(true);
+    });
+    it('should throw error for string with spaces only', () => {
+      return serverless.variables
+        .getValueStrToBool('strToBool( )')
```

Comment:
This if I see correctly will never end in resolver as it doesn't match regex.

Either's there's no point in testing it, or regex should be updated to accept empty strings (or string padding)

Reply:
Good point.